src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -294,21 +263,26 @@ private EntityTypeColumn copyColumn(EntityTypeColumn column, EntityType entityTy
     }
   }
 
-  private long countDbSources(EntityType entityType) {
-    return entityType
-      .getSources()
-      .stream()
-      .filter(source -> !Boolean.TRUE.equals(source.getFlattened()) && DB_TYPE.equals(source.getType()))
-      .count();
+  private static EntityTypeSource copySource(EntityTypeSource sourceFromParent, EntityTypeSource source, Map<String, String> renamedAliases, boolean renameAliases) {
+    return new EntityTypeSource()
+      .type(source.getType())
+      .id(source.getId())
+      .flattened(source.getFlattened())
+      .alias(renameAliases ? renamedAliases.get(source.getAlias()) : source.getAlias())
+      .target(source.getTarget())
+      .join(source.getJoin() == null ? null : new EntityTypeSourceJoin()
+        .type(source.getJoin().getType())
+        .condition(source.getJoin().getCondition())
+        .joinTo(renameAliases ? renamedAliases.get(source.getJoin().getJoinTo()) : source.getJoin().getJoinTo())
+      )
+      .useIdColumns(sourceFromParent == null || Boolean.TRUE.equals(source.getUseIdColumns()));
   }
 
-  private List<EntityTypeColumn> getFilteredColumns(List<EntityTypeColumn> unfilteredColumns) {
+  private Stream<EntityTypeColumn> getFilteredColumns(Stream<EntityTypeColumn> unfilteredColumns) {
```

Comment:
Personal opinion: We should remove this method from this class, since it's not at all related to entity type flattening. But that's out of scope of this already large PR, so I didn't bother